internal/controller/resource_manager.go

```diff
@@ -278,3 +331,65 @@ func removeListSuffix(s string) string {
 
 	return s
 }
+
+// startWatcher (re)initializes the Kubernetes watch stream using the last known resourceVersion.
+func (r *ResourceManager) startWatcher(ctx context.Context, apiGroup string, lastKnownRV string, watchOptions metav1.ListOptions, existing watch.Interface, logger *zap.Logger) (watch.Interface, error) {
+	// Always resume from the last known resourceVersion.
+	watchOptions.ResourceVersion = lastKnownRV
+	// Ensure we get periodic bookmarks for advancing the resume point.
+	watchOptions.AllowWatchBookmarks = true
+	// Stop any existing watcher before creating a new one.
+	if existing != nil {
+		existing.Stop()
+	}
+
```

Comment:
Remove this. Since you factorized the watch loop into a function (`runWatcherUntilRestart`), this can and should be done in a `defer` in that function.
```suggestion
```